static/js/apps/nl_interface/app.tsx

```diff
@@ -44,6 +45,11 @@ export function App(): JSX.Element {
   }, [queries]);
 
   function addContext(context: any, idx: number) {
+    if (isNLInterfaceNext()) {
+      // The context list is handled entirely by the server.
```

Comment:
earlier queries could be resent if the app refreshes (sometimes happens when it's low on memory etc) which is why it was always sliced to just the available context when the query was first sent. is the server also handling the index of each query?

Reply:
also, to preserve each "question/answer" component so it can be memoized, the context for each pair should be fixed. i think the append should stay the same here, but for `getContextHistory`, it should be a `get` instead of `slice`.